lib/watcher.js

```diff
@@ -303,19 +239,99 @@ export const setOptions = options => {
     retryCount = options.retryCount || 0;
 };
 
-export const addTest = (file, suiteIndex = -1) => {
-    pendingTests.push({file, suiteIndex});
+export const addTest = (file) => {
     testsList.push(file);
 };
 
+const reQueueSuite = (suite) => {
+    pendingSuites.push(suite);
+};
+
+const buildTestSuiteList = (options, throttledCalls) => {
+    // Flatten list of test suites
+    return flatten(testsList.map(file => {
+        const relativeFile = path.relative(process.cwd(), file);
+        const { mocha, beforeTestsRunCacheStateMark } = loadMochaSuite(relativeFile);
+        const suites = mocha.suite.suites;
+
+        debugLog(`Suites length is ${suites.length} for ${file}`);
+
+        // files without test suites have zero length of `mocha.suite.suites` array
+        if(suites && suites.length && suites.length > 0) {
+            return range(suites.length).map(index => buildTestSuite(file, relativeFile, index, options, throttledCalls, beforeTestsRunCacheStateMark));
```

Comment:
Turns out you can't map over generator iterables, so I used https://www.npmjs.com/package/range instead.